packages/react/src/app-layout/AppLayoutSidebarNav.tsx

```diff
@@ -109,24 +130,32 @@ function AppLayoutSidebarNavItemInner({
 	return (
 		<li
 			css={{
-				' a': {
+				'> a': {
 					textDecoration: 'none',
 				},
 
-				' a, button': {
-					display: 'flex',
-					alignItems: 'center',
-					gap: mapSpacing(1),
+				'> a, > button, > span': {
 					width: '100%',
 					boxSizing: 'border-box',
-					paddingLeft: mapSpacing(1.5),
-					paddingRight: mapSpacing(1.5),
+					wordBreak: 'break-word', // Prevent long labels from causing overflow
 					paddingTop: mapSpacing(1),
 					paddingBottom: mapSpacing(1),
+					paddingLeft: mapSpacing(1.5),
+					paddingRight: mapSpacing(1.5),
+
+					'& > svg': {
+						flexShrink: 0,
+					},
+				},
+
+				'> a, > button': {
+					position: 'relative',
+					display: 'flex',
+					alignItems: 'center',
+					gap: mapSpacing(0.75),
```

Comment:
This line re-applies changes from PR https://github.com/steelthreads/agds-next/pull/1348